File: Selenium/Project/Activity9.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.junit.jupiter.api.Assertions;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;
import java.util.Arrays;
import java.util.Iterator;
import java.util.List;
import java.util.Set;

public class Activity9 {
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void setup() {
        WebDriverManager.firefoxdriver().setup();
        driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/jobs/wp-admin");
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.manage().timeouts().implicitlyWait(Duration.ofMillis(5000));
    }

    @Test
    public void createJobListing() throws InterruptedException {

        WebElement userTextBox = driver.findElement(By.id("user_login"));
        WebElement userPassBox = driver.findElement(By.id("user_pass"));
        WebElement submitButton = driver.findElement(By.id("wp-submit"));

        userTextBox.sendKeys("root");
        userPassBox.sendKeys("pa$$w0rd");
        submitButton.click();

        WebElement jobListings = driver.findElement(By.xpath("//li[@id='menu-posts-job_listing']/a/div[3]"));
        jobListings.click();

        WebElement jobsCount = driver.findElement(By.xpath("//a[@class='current']/span"));

        String numJobsB4Add=jobsCount.getText();

        int oldJobs = Integer.parseInt(numJobsB4Add.substring(numJobsB4Add.indexOf("(")+1, numJobsB4Add.indexOf(")")));
        System.out.println("Number of Jobs before Add : " + oldJobs);

        WebElement addNew = driver.findElement(By.xpath("//div[@id='wpbody-content']/div[4]/a"));
        addNew.click();

        Thread.sleep(2000);

        WebElement modal = driver.findElement(By.xpath("//button[@aria-label='Close dialog']"));
        modal.click();

        WebElement jobPosition = driver.findElement(By.xpath("//textarea[@id='post-title-0']"));
        jobPosition.sendKeys("Data Analyst");

        WebElement jobLocation = driver.findElement(By.id("_job_location"));
        jobLocation.sendKeys("USA");

        WebElement companyName = driver.findElement(By.id("_company_name"));
        companyName.sendKeys("IBM");

        WebElement publishButton = driver.findElement(By.xpath("//div[@class='edit-post-header__settings']/button[2]"));
        publishButton.click();

        WebElement publishConfirmButton = driver.findElement(By.xpath("//div[@class='editor-post-publish-panel']/div/div/button"));
        publishConfirmButton.click();
        Thread.sleep(2000);
        driver.navigate().back();

        WebElement jobListingsConfirm = driver.findElement(By.xpath("//li[@id='menu-posts-job_listing']/a/div[3]"));
        jobListingsConfirm.click();

        WebElement newjobsText = driver.findElement(By.xpath("//a[@class='current']/span"));
        String newJobsCountText=newjobsText.getText();
        int newJobsCount = Integer.parseInt(newJobsCountText.substring(newJobsCountText.indexOf("(")+1, newJobsCountText.indexOf(")")));
        System.out.println("Number of Jobs After Add : " + newJobsCount);

        Assertions.assertEquals(oldJobs+1, newJobsCount);
        Thread.sleep(2000);
    }

    @AfterClass
    public void cleanup() {
        driver.quit();
    }
}

File: Selenium/Project/Activity9.py
# Import required packages from selenium
import time

import pytest
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait


class TestClass():

    def test_create_job_listings(self, my_fixture):
        # Initialise Driver
        driver = my_fixture
        url = 'https://alchemy.hguy.co/jobs/wp-admin'
        driver.get(url)

        # Initialise Wait for 10 secs
        wait = WebDriverWait(driver, 10)

        # Find user/password textbox and login
        userTextBox = driver.find_element(By.ID, "user_login")
        userPassBox = driver.find_element(By.ID, "user_pass")
        submitButton = driver.find_element(By.ID, "wp-submit")

        userTextBox.send_keys("root")
        userPassBox.send_keys("pa$$w0rd")
        submitButton.click()

        # Get the job listings
        jobListings = driver.find_element(By.XPATH, "//li[@id='menu-posts-job_listing']/a/div[3]")
        jobListings.click()

        # Get the count of job listings and print the count
        jobListings = driver.find_element(By.XPATH, "//a[@class='current']/span")
        jobsCount = ''.join(filter(lambda i: i.isdigit(), jobListings.text))
        print("Number of Jobs before Add : " + jobsCount)

        # Add a new Job to the List
        addNew = driver.find_element(By.XPATH, "//div[@id='wpbody-content']/div[4]/a")
        addNew.click()
        time.sleep(2)

        # Fill up the New Job form
        modal = driver.find_element(By.XPATH, "//button[@aria-label='Close dialog']")
        modal.click()

        jobPosition = driver.find_element(By.XPATH, "//textarea[@id='post-title-0']")
        jobPosition.send_keys("Data Analyst")

        jobLocation = driver.find_element(By.ID, "_job_location")
        jobLocation.send_keys("USA")

        companyName = driver.find_element(By.ID, "_company_name")
        companyName.send_keys("IBM")

        # Publish the new job Posting
        publishButton = driver.find_element(By.XPATH, "//div[@class='edit-post-header__settings']/button[2]")
        publishButton.click()

        publishConfirmButton = driver.find_element(By.XPATH, "//div[@class='editor-post-publish-panel']/div/div/button")
        publishConfirmButton.click()
        time.sleep(2)
        driver.back()

        jobListingsConfirm = driver.find_element(By.XPATH, "//li[@id='menu-posts-job_listing']/a/div[3]")
        jobListingsConfirm.click()

        # Get the count of job listings and print the count
        jobListings = driver.find_element(By.XPATH, "//a[@class='current']/span")
        newJobsCount = ''.join(filter(lambda i: i.isdigit(), jobListings.text))
        print("Number of Jobs before Add : " + newJobsCount)

        assert int(newJobsCount) == int(jobsCount) + 1
        time.sleep(2)
